src/hono-base.ts

```diff
@@ -65,7 +65,7 @@ class Hono<E extends Env = Env, S = {}, BasePath extends string = '/'> extends d
     To use it, inherit the class and implement router in the constructor.
   */
   router!: Router<H>
-  readonly getPath: (request: Request) => string
+  readonly getPath: (request: Request, options?: { env?: E['Bindings'] }) => string
```

Comment:
Due to this type declaration I have to write `const env = options?.env ?? ({} as Env)` in `getPath` to get the bindings which feels a bit hacky/unnecessary. 

I tried changing this to:
```ts
readonly getPath: ((request: Request) => string) | ((request: Request, options: { env: E['Bindings'] }) => string)
```

So, basically some kind of overloading instead of an optional param. However, this didn't work out well:
- when declaring `getPath` as `getPath: (req) => { ... }` I get `Parameter 'req' implicitly has an 'any' type.` (i.e. this change would break backward compatibility)
- when declaring `getPath` as `getPath: (req, { env }) => { ... }` the `env` variable still has an optional type for whatever reason (i.e. `Bindings | undefined`)

I'm not super proficient in TypeScript, so I'm not exactly sure what's going on. Do you have any other idea on whether it'd be possible to make type declarations more user-friendly without introducing breaking changes?

Reply:
Yeah, sometimes TypeScript can be troublesome.

In this case, it's preferable to keep backward compatibility with my implementation. I believe using `options!.env!` or `options?.env?` isn't too hacky. It simply translates to:

```ts
if (options && options.env) {
  //...
}
```

But thank you for the suggestion!